contracts/Sacd.sol

```diff
@@ -96,12 +92,39 @@ contract Sacd is ISacd, Initializable, AccessControlUpgradeable, UUPSUpgradeable
         revert ZeroAddress();
       }
 
+      // Validate template permissions if template is used
       SacdStorage storage $ = _getSacdStorage();
 
-      uint256 tokenIdVersion = $.tokenIdToVersion[asset][tokenId];
-      $.permissionRecords[asset][tokenId][tokenIdVersion][grantee] = PermissionRecord(permissions, expiration, source);
+      if (templateId != 0) {
+        address templateContract = $.templateContract;
+
+        if (templateContract != address(0)) {
+          try ITemplate(templateContract).getTemplate(templateId) returns (ITemplate.TemplateData memory template) {
+            if (!template.isActive) {
+              revert TemplateNotActive(templateId);
+            }
+            if (template.asset != asset) {
+              revert TemplateAssetMismatch(templateId, template.asset, asset);
+            }
+            if ((template.permissions & permissions) != template.permissions) {
```

Comment:
I thought we said that these had to match exactly, in which case `permissions == template.permissions` would be enough. We're allowing the SACD to extend the permissions in the template?

There have been so many conversations that I can't keep it straight.

Reply:
Maybe we should just get rid of the prefix and just check the `Qm` in the CID. I can't remember the reason for this prefix, but I can't see how it helps